src/components/CreateGroupDialog.vue

```diff
@@ -33,15 +33,27 @@
 import { Modal } from "bootstrap";
 
 export default {
+    beforeRouteLeave(to, from, next) {
+        this.cleanupModal();
+        next();
+    },
     props: {},
     emits: [ "added" ],
     data: () => ({
         modal: null,
         groupName: null,
     }),
+    watch: {
+        $route(to, from) {
+            this.cleanupModal();
+        }
+    },
```

Comment:
is this not redundant to `beforeRouteLeave`?

```suggestion
```

Reply:
hmm yea i was trying out both ways seeing which one worked then just left both off them there. beforeRouteLeave should handle the modal cleanup during route transitions then will remove any other redundant code